common/models/role.js

```diff
@@ -29,42 +29,55 @@ module.exports = function(Role) {
   // Set up the connection to users/applications/roles once the model
   Role.once('dataSourceAttached', function() {
     var roleMappingModel = this.RoleMapping || loopback.getModelByType(RoleMapping);
+
+    /**
+     * Fetch the ids of all users assigned to this role
+     * @param  {Function} callback
+     */
     Role.prototype.users = function(callback) {
-      roleMappingModel.find({where: {roleId: this.id,
-        principalType: RoleMapping.USER}}, function(err, mappings) {
+      roleMappingModel.find({
+        where: {roleId: this.id, principalType: RoleMapping.USER}
+      }, function(err, mappings) {
         if (err) {
-          if (callback) callback(err);
-          return;
+          return callback(err);
         }
-        return mappings.map(function(m) {
+        callback(null, mappings.map(function(m) {
           return m.principalId;
```

Comment:
I considered that, but there were concerns around the potential size of the document so I decided to follow the original idea of returning the id. What do you think about accepting a fields param as the second argument and passing it to the underlying `find` so that the user isn't forced to load the entire document every time they use this function?

```
Role.prototype.users = function(fields, callback) {
  if (callback === undefined) {
    callback = fields;
    fields = {};
  }

  roleMappingModel.find({
    where: {roleId: this.id, principalType: RoleMapping.USER}
  }, function(err, mappings) {
    var ids;

    if (err) {
      return callback(err);
    }
    
    ids = mappings.map(function(m) {
      return m.principalId;
    });

    User.find({
      where: {id: {inq: ids}, fields: fields}
    }, callback);
  });
};
```